### PR TITLE
Increase object pkg test coverage

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Increase object pkg test coverage.

**Related issue(s)**

- #240

### CODE REVIEW

File: pkg/object/apirule.go [hunk 1]
```diff
@@ -64,19 +64,17 @@ func GetService(svcName string, port uint32) apigatewayv1beta1.Service {
 
 // WithService sets the Service of an APIRule.
 func WithService(host, svcName string, port uint32) Option {
-	return func(o metav1.Object) {
-		d := o.(*apigatewayv1beta1.APIRule)
+	return func(o *apigatewayv1beta1.APIRule) {
```
Comment (k15r) [inline pkg/object/apirule.go:67]:
```suggestion
	return func(r *apigatewayv1beta1.APIRule) {
```
nit: why o for APIRule?

File: pkg/object/apirule.go [hunk 2]
```diff
@@ -64,19 +64,17 @@ func GetService(svcName string, port uint32) apigatewayv1beta1.Service {
...
+		o.Spec.Host = &host
 	}
 }
 
 // WithGateway sets the gateway of an APIRule.
 func WithGateway(gw string) Option {
-	return func(o metav1.Object) {
-		d := o.(*apigatewayv1beta1.APIRule)
-		d.Spec.Gateway = &gw
+	return func(o *apigatewayv1beta1.APIRule) {
```
Comment (k15r) [inline pkg/object/apirule.go:76]:
```suggestion
	return func(r *apigatewayv1beta1.APIRule) {
```

File: pkg/object/apirule.go (hunk 1 above)
Comment (marcobebway) [reply pkg/object/apirule.go]: Done

File: pkg/object/apirule.go (hunk 2 above)
Comment (marcobebway) [reply pkg/object/apirule.go]: Done

**Review by k15r — APPROVED** [review]

**Pull request merged**